File: sdks/typescript/src/sinks/ConsoleSink.ts
import { once } from 'node:events';

import type { Sink } from './Sink.js';
import type { WideEventPayload } from '../types.js';

export class ConsoleSink implements Sink {
  async export(event: WideEventPayload): Promise<void> {
    if (!process.stdout.write(`${JSON.stringify(event)}\n`)) {
      await once(process.stdout, 'drain');
    }
  }
}

File: sdks/python/src/openakta_logger/sinks/console_sink.py
from __future__ import annotations

import asyncio
import json
import sys
from typing import Any

from .base import Sink


class ConsoleSink(Sink):
    async def export(self, event: dict[str, Any]) -> None:
        line = json.dumps(event, separators=(",", ":"), ensure_ascii=False) + "\n"
        await asyncio.to_thread(sys.stdout.write, line)
